### PR TITLE
Added popular function to API

### PR DESCRIPTION
**Pull request by Jayy001**
* Blacked everything
* Added popular API

Had some issues with getting this, I wanted it to remove all the hash types from the dict if it wasn't in the popular list but for some reason it would only remove a few, yet, if I did it the opposite way (making a new list and then appending) ones, it worked fine. Drove me nuts for a bit. Couldn't find any other cleaner way :(. - If you knew why it was doing this please tell me! :)

### CODE REVIEW

File: name_that_hash/prettifier.py [hunk 1]
```diff
@@ -20,15 +21,17 @@ def __init__(self, kwargs, api=False):
             self.a11y = kwargs["accessible"]
             self.john = kwargs["no_john"]
             self.hashcat = kwargs["no_hashcat"]
+        self.args = kwargs
+        self.hashinfo_obj = hash_info.HashInformation()
```
Comment (bee-san) [inline name_that_hash/prettifier.py:25]: ??? this indenting bruh

File: name_that_hash/prettifier.py [hunk 2]
```diff
@@ -20,15 +21,17 @@ def __init__(self, kwargs, api=False):
...
         """
-        takes the prototypes and turns it into json
-        returns the json
+		takes the prototypes and turns it into json
+		returns the json
 
-        Doesn't print it, it prints in main
-        """
+		Doesn't print it, it prints in main
+		"""
```
Comment (bee-san) [inline name_that_hash/prettifier.py:34]: indenting??

File: name_that_hash/prettifier.py [hunk 3]
```diff
@@ -37,21 +40,30 @@ def turn_hash_objs_into_dict(self, objs: List):
...
             logging.debug(f"Output_as_dicts is now {outputs_as_dict}")
+
+        if self.args["popular_only"]:
+            popular_only = {}
+            for hash in list(outputs_as_dict.keys()):
+                popular_only[hash] = []
+                for hash_type in outputs_as_dict[hash]:
+                    if hash_type["name"] in self.hashinfo_obj.popular:
+                        popular_only[hash].append(hash_type)
+            return popular_only
```
Comment (bee-san) [inline name_that_hash/prettifier.py:51]: tbh this should be a separate function

File: name_that_hash/prettifier.py [hunk 4]
```diff
@@ -37,21 +40,30 @@ def turn_hash_objs_into_dict(self, objs: List):
...
+		1.
+		2.
+		3.
+		4.
 
 
-        then everything else on one line.
-        """
+		then everything else on one line.
+		"""
```
Comment (bee-san) [inline name_that_hash/prettifier.py:66]: Indenting issues

File: tests/test_click.py [hunk 5]
```diff
@@ -68,4 +68,3 @@ def test_kerberos_works():
         ],
     )
```
Comment (bee-san) [inline tests/test_click.py:69]: You need tests for most popular hashes :-) Call both the regex and click module to make sure it works 100%

**Review by bee-san — CHANGES_REQUESTED** [review]
Looking good! Left some comments. Also:
1. You need tests (both click and function)
2. API documentation needs to be updated for this :-)

File: tests/test_click.py (hunk 5 above)
Comment (Jayy001) [reply tests/test_click.py:69]: 👍🏻

**Comment by Jayy001** [comment]
Should be all good :)

File: name_that_hash/prettifier.py [hunk 6]
```diff
@@ -37,21 +45,33 @@ def turn_hash_objs_into_dict(self, objs: List):
         for y in objs:
             outputs_as_dict.update(y.hash_obj)
             logging.debug(f"Output_as_dicts is now {outputs_as_dict}")
+
+        if self.args["popular_only"]:
```
Comment (bee-san) [inline name_that_hash/prettifier.py:49]: docs? i have no idea how to use this :(

**Review by bee-san — CHANGES_REQUESTED** [review]
I also asked for `click` tests :)

**Comment by bee-san** [comment]
Lol I tried to fix your merge conflicts and it broke everything sorry

**Comment by bee-san** [comment]
@amadejpapez any chance you can fix the merge conflicts on this please? I broke it 😢

File: name_that_hash/hashes.py [hunk 7]
```diff
@@ -1611,6 +1611,29 @@ class HashInfo:
             r"^\$krb5asrep\$23\$[^:]+:[a-f0-9]{32,32}\$[a-f0-9]{64,40960}$",
```
[the remark this answers is not in the document]
Comment (bee-san) [reply name_that_hash/hashes.py:1611]: this didnt work for me

[the remark this answers is not in the document]
Comment (bee-san) [reply name_that_hash/hashes.py:1611]: nvm

**Pull request merged**